src/libutil/cf.c

```diff
@@ -210,6 +215,121 @@ bool cf_array_contains (const cf_t *cf, const char *str)
     return false;
 }
 
+static bool parent_dir_is_secure (cf_t *cf,
+                                  const char *filename,
+                                  struct cf_error *error)
+{
+    struct stat st;
+    char buf [PATH_MAX + 1];
+    char *dir;
+
+    if (strlcpy (buf, filename, sizeof (buf)) >= sizeof (buf)
+        || !(dir = dirname (buf))) {
+        errno = ENAMETOOLONG;
+        errprintf (error, filename, -1, "Unable to get dirname");
+        return false;
+    }
+    if (lstat (dir, &st) < 0) {
+        errprintf (error, filename, -1, "Unable to stat parent directory");
+        return false;
+    }
+    if (!S_ISDIR (st.st_mode)) {
+        errprintf (error, filename, -1,
+                   "Unable to check parent directory. Unexpected file type");
+        errno = EINVAL;
+        return false;
+    }
+    if ((st.st_uid != 0)
+        && (st.st_uid != geteuid ())) {
+        errprintf (error, filename, -1,
+                   "Invalid ownership on parent directory");
+        errno = EINVAL;
+        return false;
+    }
+    if (st.st_gid != 0
+        && st.st_gid != getegid ()
+        && (st.st_mode & S_IWGRP)
+        && !(st.st_mode & S_ISVTX)) {
+        errprintf (error, filename, -1,
+                   "parent directory is group-writeable without sticky bit");
+        errno = EINVAL;
+        return false;
+    }
```

Comment:
So sys admins would need to ensure that config tree for flux-security is all `root:root`.  `flux:flux` would not be appropriate since flux isn't supposed to have any special privileges.  Now that I state that out loud it seems obvious... Duh.

The other allowed case would be for testing where some user owns everything and runs without setuid.  Got it.